docs/csharp/language-reference/compiler-messages/cs1983.md

```diff
@@ -0,0 +1,47 @@
+---
+description: "Compiler Error CS1983"
+title: "Compiler Error CS1983"
+ms.date: 9/12/2022
+f1_keywords:
+  - "CS1983"
+helpviewer_keywords:
+  - "CS1983"
+---
+# Compiler Error CS1983
+
+The return type of an async method must be void, Task, Task\<T\>, a task-like type, IAsyncEnumerable\<T\>, or IAsyncEnumerator\<T\>
+
+## Example
+
+ The following sample generates CS1983:
+
+```csharp
+// CS1983.cs (4,62)
+using System.Collections.Generic;
+
+class C
+{
+    static async IEnumerable<int> M()
+    {
+        yield return await Task.FromResult(1);
+    }
+}
+```
+
+Since  `IEnumerable.GetEnumerator` returns an `IEnumerator<T>` whose `MoveNext` method does not return a value of `Type<T>`, it is not compatible with an `async` method.
```

Comment:
typo:

```suggestion
Since  `IEnumerable.GetEnumerator` returns an `IEnumerator<T>` whose `MoveNext` method does not return a value of `Task<T>`, it is not compatible with an `async` method.
```